src/Core/src/WindowOverlay/WindowOverlay.Android.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Android.App;
+using Android.Views;
+using AndroidX.CoordinatorLayout.Widget;
+using Microsoft.Maui.Graphics;
+using Microsoft.Maui.Graphics.Native;
+using Microsoft.Maui.Handlers;
+
+namespace Microsoft.Maui
+{
+	public partial class WindowOverlay : IWindowOverlay, IDrawable
+	{
+		internal Activity? _nativeActivity;
+		internal NativeGraphicsView? _graphicsView;
+		ViewGroup? _nativeLayer;
+
+		/// <inheritdoc/>
+		public bool DisableUITouchEventPassthrough { get; set; }
+
+		public bool InitializeNativeLayer()
+		{
+			if (IsNativeViewInitialized)
+				return true;
+
+			if (Window == null)
+				return false;
+
+			var nativeWindow = Window.Content.GetNative(true);
+			if (nativeWindow == null)
+				return false;
+			
+			var handler = Window.Handler as WindowHandler;
+			if (handler == null || handler.MauiContext == null)
+				return false;
+			var rootManager = handler.MauiContext.GetNavigationRootManager();
+			if (rootManager == null)
+				return false;
+
+
+			if (handler.NativeView is not Activity activity)
+				return false;
+
+			_nativeActivity = activity;
+			_nativeLayer = rootManager.RootView as ViewGroup;
+
+			if (_nativeLayer == null || _nativeLayer.Context == null)
+				return false;
+
+			if (_nativeActivity == null || _nativeActivity.WindowManager == null || _nativeActivity.WindowManager.DefaultDisplay == null)
```

Comment:
It's the use of "nullable" enforcing these checks, but I had started using some of the individual values for each when I had started writing that code out, so it would have made more sense. I can go back and clean these up.